lxd/storage/drivers/driver_zfs_volumes.go

```diff
@@ -1821,3 +1829,13 @@ func (d *zfs) RenameVolumeSnapshot(vol Volume, newSnapshotName string, op *opera
 
 	return nil
 }
+
+// GetGetVolumeSize returns a volume's size.
+func (d *zfs) GetVolumeSize(vol Volume) (int64, error) {
+	size, err := d.getDatasetProperty(d.dataset(vol, false), "available")
```

Comment:
This doesn't feel like the right properly for this.

If you want the max size, you want quota/refquota (container) or volsize (VM).
If you want the currently used size, then you want referenced (container) or volsize (VM).